c10/util/Half.h

```diff
@@ -426,6 +426,28 @@ struct alignas(4) complex<Half> {
   constexpr C10_HOST_DEVICE Half imag() const {
     return imag_;
   }
+
+  complex<Half>& operator+=(const complex<Half>& other) {
```

Comment:
these can't be constexpr as `c10::Half(float val)` is not constexpr

Reply:
and why is that?

Reply:
It is defined like this and depends on device specify helpers and CPU path uses `memcpy`. 
https://github.com/pytorch/pytorch/blob/6fd14ba9db3301e74b7e654449c5f3409564772c/c10/util/Half-inl.h#L28-L36

Reply:
I see. can you leave an inline comment about it?